climada/engine/impact.py

```diff
@@ -390,6 +388,44 @@ def impact_per_year(self, all_years=True, year_range=None):
             year_set[year] = sum(self.at_event[orig_year == year])
         return year_set
 
+    def impact_at_reg(self, agg_regions=None):
+        """Aggregate impact on given aggregation regions. This method works 
+        only if Impact.imp_mat was stored during the impact calculation.
+
+        Parameters
+        ----------
+        agg_regions : np.array, list, pd.Series
+            It has the same lenght as the number of coordinates points in exposure 
+            and reports what macro-regions these points belong to. If no aggregation
+            regions are passed, the method aggregates impact at the admin_0 level.
+            Default is None.
+
+        Returns
+        -------
+        pd.DataFrame
+        """
+        if self.imp_mat.nnz == 0:
+            LOGGER.warning("No Impact.imp_mat was stored during the impact calculation,"
+                           "thus the impact per region cannot be computed")
+            return None
```

Comment:
Could also return an empty dataframe. Not sure what would be smartest here. I personally would prefer throwing an error because the user should be aware that using the method in this context makes no sense.

Reply:
agree on throwing error. done.